src/pick.c

```diff
@@ -612,33 +614,54 @@ print_choices(int selection)
 		if (i == selection)
 			tty_putp(enter_standout_mode);
 
+		esc = 0;
+		p = 0;
+
 		for (j = 0, length = MIN(choice->match_start, (size_t)columns);
 		     j < (ssize_t)length;
-		     j++)
+		     j++) {
+			if (esc)
+				esc = IN_ESC_SEQ(choice->string, j);
+			else if (!(esc = (choice->string[j] == ESCAPE)))
+				p++;
```

Comment:
Maybe these could be extracted like:

```c
p = skip_esc(p, choice->string, j, &esc);
```

So I guess:

```c
int
skip_esc(int p, char *str, int j, int *esc)
{
  if (*esc) {
    *esc = IN_ESC_SEQ(str, j);
    return p;
  } else if(!(*esc = (str[j] == ESCAPE)))
    return p + 1;
}
```

I don't love that `(!(esc = (choice->string[j] == ESCAPE)))` line because I was skimming for where `esc` is set and instead I have to carefully read this line. How about pulling out the assignment from the comparison?